src/converters/app-converter.ts

```diff
@@ -0,0 +1,388 @@
+import * as pulumi from '@pulumi/pulumi';
+import { AssemblyManifestReader, StackManifest } from '../assembly';
+import { ConstructInfo, GraphBuilder } from '../graph';
+import { IStackComponent, lift, Mapping } from '../types';
+import { ArtifactConverter, FileAssetManifestConverter } from './artifact-converter';
+import { CdkConstruct, ResourceMapping } from '../interop';
+import { debug } from '@pulumi/pulumi/log';
+import {
+    cidr,
+    getAccountId,
+    getAzs,
+    getRegion,
+    getSsmParameterList,
+    getSsmParameterString,
+    getUrlSuffix,
+} from '@pulumi/aws-native';
+import { mapToAwsResource } from '../aws-resource-mappings';
+import { attributePropertyName, mapToCfnResource } from '../cfn-resource-mappings';
+import { CloudFormationResource, getDependsOn } from '../cfn';
+import { OutputMap, OutputRepr } from '../output-map';
+import { parseSub } from '../sub';
+
+/**
+ * AppConverter will convert all CDK resources into Pulumi resources.
+ */
+export class AppConverter {
+    // Map of stack artifactId to StackConverter
+    public readonly stacks = new Map<string, StackConverter>();
+
+    public readonly manifestReader: AssemblyManifestReader;
+
+    constructor(readonly host: IStackComponent) {
+        this.manifestReader = AssemblyManifestReader.fromPath(host.assemblyDir);
+    }
+
+    convert() {
```

Comment:
If I've been following right convert() methods actually issue Pulumi resource calls e.g. `new Bucket()`.

Reply:
Yes I think that is correct.